client.go

```diff
@@ -54,7 +55,11 @@ type VaultTokenInfo struct {
 	Type           string      `json:"type"`
 }
 
-// NewClient returns a new client based on the provided config
+// NewClient returns a new client based on the provided config.
+//
+// If you are using the client and then disposing it, make sure you call
+// `defer client.Shutdown()` once you are done with it. Failing to do so will
```

Comment:
Do backticks render properly in godoc?

Feel free to use/ignore my suggestion as needed.

```suggestion
// defer client.Shutdown() once you are done with it. Failing to do so will
```

Reply:
they do not, I believe we added just as a visual clue. Happy to remove it